src/sentry/event_manager.py

```diff
@@ -519,11 +616,53 @@ def stringify(f):
         if data['transaction']:
             data['transaction'] = trim(data['transaction'], MAX_CULPRIT_LENGTH)
 
-        return data
+        self._data = data
+
+    def should_filter(self, project=None, client_ip=None):
+        """
+        returns (result: bool, reason: string or None)
+        Result is True if an event should be filtered
+        The reason for filtering is passed along as a string
+        so that we can store it in metrics
+        """
+        for name in SECURITY_REPORT_INTERFACES:
+            if name in self._data:
+                interface = get_interface(name)
+                if interface.to_python(self._data[name]).should_filter(project):
```

Comment:
From what I can see, this can only return true for the `Csp` interface. We inline that logic, remove the loop, and even remove that `should_filter` method alltogether, to reduce dependencies on the interface code.

Reply:
I don't understand what you mean by that. We can't remove `should_filter` from EventManager

Reply:
I meant removing `should_filter` from the interfaces. All of them return `False` statically, except the `Csp` interface. And that could could be inlined here.

Reply:
Ah yes. That was somewhere on my radar but I'd like to deploy in smaller steps